Neuro/Minigames/Solvers/CleanO2FilterSolver.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections;
+using UnityEngine;
+using Neuro.Cursor;
+using System.Linq;
+
+namespace Neuro.Minigames.Solvers;
+
+[MinigameSolver(typeof(LeafMinigame))]
+public class CleanO2FilterSolver : MinigameSolver<LeafMinigame>
+{
+    protected override IEnumerator CompleteMinigame(LeafMinigame minigame, NormalPlayerTask task)
+    {
+        // ValidArea is somewhere offscreen so easier to just move towards the arrows on the chute
+        Vector3 exit = minigame.Arrows[0].transform.position;
+        Collider2D lastLeaf = null;
+        while (!task.IsComplete)
+        {
+            Collider2D _lastLeafCopy = lastLeaf;
+            Collider2D leaf = minigame.Leaves
+                .Where(l => l && l.enabled && l.gameObject.active && l != _lastLeafCopy
+                    && Vector3.Distance(l.transform.position, exit) > 4f)  
+                .MaxBy(l => (InGameCursor.Instance.Position - (Vector2)l.transform.position).magnitude);
```

Comment:
Is there a need for `Vector3.Distance(l.transform.position, exit) > 4f)` if you are already getting the furthest one with `.MaxBy`?

Reply:
I ran into issues with the cursor trying to grab a leaf that was already being sucked in / destroyed, causing null errors even after the null check. Forcing the cursor to start away from the vent fixes that issue.